### PR TITLE
refactor: state for wizard

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

Fixes wizard state

## Issue

#2193 #2308 

## How to test

1. Goto Create Agreement wizard
2. complete step 1 and 2
3. make a budgetline on step 3
4. go back to step 2
5. go back to step 3
6. 🟢 budget-lines should persist (state is coming from wizard)

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: frontend/src/components/UI/RadioButtonTile/RadioButtonTile.test.js [hunk 1]
```diff
@@ -24,14 +27,16 @@ describe("RadioButtonTile", () => {
         );
         const input = screen.getByRole("radio");
         fireEvent.click(input);
-        expect(input).toBeChecked();
+        input.checked = true; // manually set the checked property
```
Comment (fpigeonjr) [inline frontend/src/components/UI/RadioButtonTile/RadioButtonTile.test.js:30]: `happy-dom` was not happy about this

File: frontend/src/pages/agreements/CreateEditAgreement.jsx [hunk 2]
```diff
@@ -76,11 +76,12 @@ export const CreateEditAgreement = ({ budgetLines, setAgreementId = () => {} })
             cancelMsg = cancelMessages.isCreatedMode;
             break;
     }
-
-    // NOTE: Consider refactoring to elevate handleCancel to this parent component
+    const handleFinish = (formData) => {
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/CreateEditAgreement.jsx:79]: could be used for api calls when completing the wizard

File: frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js [hunk 3]
```diff
@@ -64,9 +66,21 @@ const useCreateBLIsAndSCs = (
     const [addBudgetLineItem] = useAddBudgetLineItemMutation();
     const [deleteBudgetLineItem] = useDeleteBudgetLineItemMutation();
     const loggedInUserFullName = useGetLoggedInUserFullName();
-    const feesForCards = getProcurementShopSubTotal(selectedAgreement, budgetLines);
+    const feesForCards = getProcurementShopSubTotal(selectedAgreement, tempBudgetLines);
     const subTotalForCards = budgetLinesTotal(budgetLines);
-    const totalsForCards = subTotalForCards + getProcurementShopSubTotal(selectedAgreement, budgetLines);
+    const totalsForCards = subTotalForCards + getProcurementShopSubTotal(selectedAgreement, tempBudgetLines);
+
+    React.useEffect(() => {
```
Comment (fpigeonjr) [inline frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js:73]: 🤷

Comment (fpigeonjr) [reply frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js:73]: use budgetlines if there are any
else use wizard state (formData)
lastly use empty array

**Review by stevtek — APPROVED** [review]

File: frontend/src/pages/agreements/CreateAgreementFlow.jsx [hunk 4]
```diff
@@ -1,20 +1,39 @@
 import React from "react";
 
-export const CreateAgreementFlow = ({ children, onFinish }) => {
+/**
+ * This is a React component that handles the flow of creating an agreement.
```
Comment (maiyerlee) [inline frontend/src/pages/agreements/CreateAgreementFlow.jsx:4]: +1 for the function documentation! Horray!

**Review by maiyerlee — APPROVED** [review]

File: frontend/src/pages/agreements/CreateEditAgreement.jsx (hunk 2 above)
Comment (fpigeonjr) [reply frontend/src/pages/agreements/CreateEditAgreement.jsx:79]: will create a tech debt item for this

**Pull request merged**